sourcegraph/sourcegraph
# vsce: vary search pattern type depending on version number

Closes https://github.com/sourcegraph/sourcegraph/issues/41236

Changes search query `patternType` param value from *standard* to *literal* on Sourcegraph versions older than 3.43.0 as they don't support *standard* pattern type value ([source](https://docs.sourcegraph.com/CHANGELOG#3-43-0)).

See also https://github.com/sourcegraph/sourcegraph/pull/42105

## Test plan
Manually tested:
- made `sourcegraph.url` VSCode setting point to Sourcegraph instance v3.42 or older
- submitted a search term and ensure that it resolves without `Unrecognized patterType` error



## App preview:

- [Web](https://sg-web-taras-yemets-vsce-fix-pattern.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-ygrlomnewv.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
